Sync with main 2022-1-30

nunnatsa opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
```

nirarg commented on pkg/managers/vm/machine_scope.go:
```diff
@@ -494,16 +495,14 @@ func (s *machineScope) setProviderStatus(vm *kubevirtapiv1.VirtualMachine, vmi *
-			return err
+	klog.V(5).Infof("using hostname %s to resolve addresses", vm.Name)
+	ips, err := net.LookupIP(vm.Name)
+	if err == nil {
```
Don't we want, in case there is error, to return the error?
```
if err != nil {
   return err
}
```

ravidbro replied:
I'm not sure about that since it takes some time since the VM was created and until the name is resolvable and this method is called as part of the create method, which I don't think we want to fail.
It's the same as it is now with guest-agent were it take some time until the IP is updated on the VMI, and until then, the code is ignoring that (see https://github.com/openshift/cluster-api-provider-kubevirt/blob/master/pkg/managers/vm/utils.go#L108)

nirarg replied:
But what would happen in this case, if not returning error?
What do we expect to happen in case we don't have the IPs?

ravidbro replied:
I think that the worker's CSR won't be approved until the IP will be updated, but it's the same as it happens now with guest-agent

nirarg replied:
Ok, got it

bardielle commented on pkg/managers/vm/machine_scope.go:
```diff
@@ -494,16 +495,14 @@ func (s *machineScope) setProviderStatus(vm *kubevirtapiv1.VirtualMachine, vmi *
-		if err != nil {
-			klog.Errorf("%s: Error extracting vm IP addresses: %v", s.machine.GetName(), err)
-			return err
+	klog.V(5).Infof("using hostname %s to resolve addresses", vm.Name)
```
Please add the bug that open

ravidbro replied:
I don't think it's a bug.
I'm really not convinced that using guest-agent is better than this method, and I don't see any reason why this method should stop working someday

bardielle commented:
/lgtm

nirarg committed:
Change OWNERS file

nunnatsa committed:
Create coverage file on unit test in CI

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #59 from nunnatsa/add-coverage-file

Create coverage file on unit test in CI

davidvossel committed:
Adds basic e2e test suite and makefile targets

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Auto download e2e test dependencies

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add simple cluster creation test to replace hack/functest.sh script

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Wire e2e tests to functests.sh script for kubevirt ci

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add comment and link to issue regarding cluster tear down in e2e test

Signed-off-by: David Vossel <davidvossel@gmail.com>

rmohr committed:
Add the new --fail-with-job flag to let mkpj fail when the prowjob fails

Signed-off-by: Roman Mohr <rmohr@redhat.com>

k8s-ci-robot committed:
Merge pull request #66 from davidvossel/e2e-testsuite

e2e ginkgo/gomega test suite

agradouski committed:
0.1.x version of kubevirt provider is compatible with 1.0.x version of cluster-api

Signed-off-by: Alex Gradouski <agradouski@apple.com>

agradouski committed:
update template

Signed-off-by: Alex Gradouski <agradouski@apple.com>

k8s-ci-robot committed:
Merge pull request #67 from rmohr/mkpj-fail

Add the new --fail-with-job flag to let mkpj fail when the prowjob fails

rmohr committed:
Run e2e test also on label events

Ensure that we react to a label change on the PR to ensure that the e2e
tests get triggered if someone writes /ok-to-test some time after the PR
got created.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

k8s-ci-robot committed:
Merge pull request #71 from rmohr/ok-to-test-fix

Run e2e test also on label events

dgovinndaraju12 committed:
Add unit test for Kubevirt Cluster Controller

k8s-ci-robot committed:
Merge pull request #68 from agradouski/fix-issue-62

make 0.1.x version of kubevirt provider compatible with 1.0.x version of cluster-api

davidvossel committed:
Continually reconcile VMIs

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Unit tests to detect when VMI is no longer ready

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Should treat VM and VMI independently during reconcile creation flow

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Unit tests to ensure VM and VMI are treated independently when determining if creation is necessary

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add template and ci for testing persistent storage vms

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add e2e test validating VMs with persistent storage

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add namespace var to templates for VMs hosted on management cluster

Signed-off-by: David Vossel <davidvossel@gmail.com>

wangxin311 committed:
fix issue #74

k8s-ci-robot committed:
Merge pull request #69 from agradouski/update-templates

fix API versions in external infra template

agradouski committed:
Add rmohr to the approvers list

k8s-ci-robot committed:
Merge pull request #73 from davidvossel/persistent-vms

Continually reconcile KubeVirtMachine objects

k8s-ci-robot committed:
Merge pull request #77 from kubernetes-sigs/add-approver-rmohr

Add rmohr to the approvers list

nunnatsa committed:
Fix bug in KubevirtCluster Reconciler - endless reconcile loop with external controller

When the KubevirtCluster CR is manage by an external controller, sometimes there is an update race between capk and the external controller, where each update in one hand, triggers reconcile and update in the other hand and vice versa.

This PR changes that so in this case, the CR is filtered out from the reconciling watch, and so capk will ignore changes made by an external controller.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #75 from wangxin311/lb

fix issue #74

k8s-ci-robot committed:
Merge pull request #72 from dgovinndaraju12/controller-unit-tests

Add unit test for Kubevirt Cluster Controller

k8s-ci-robot committed:
Merge pull request #79 from nunnatsa/fix-bug-kv-cluster-race

Fix bug in KubevirtCluster Reconciler - endless reconcile loop with external controller

openshift-merge-robot committed:
Merge pull request #10 from nunnatsa/sync

Sync with upstream 2022.1.29

nirarg commented:
/lgtm
